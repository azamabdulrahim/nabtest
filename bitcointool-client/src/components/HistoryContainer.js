import React, { Component } from "react";
import { connect } from "react-redux";
import { getHistory } from "../actions/historyActions";
import PropTypes from "prop-types";
import HistoryTable from "./HistoryTable";

export class HistoryContainer extends Component {
  componentDidMount() {
    const { refresh, date, currency } = this.props;

    if (refresh && date) {
      this.props.getHistory(date, currency);
    }
  }
  render() {
    let displayResult;
    let history;

    if (this.props.history) {
      history = this.props.history.history;
    }

    if (history && history.length > 0) {
      displayResult = <HistoryTable history={history} />;
    } else {
      displayResult = <span>No data</span>;
    }

    return (
      <div>
        <p />
        <h2>Price History</h2>
        {displayResult}
      </div>
    );
  }
}

HistoryContainer.propTypes = {
  history: PropTypes.object,
  getHistory: PropTypes.func,
  errors: PropTypes.object
};

const mapStateToProps = state => ({
  history: state.history,
  errors: state.errors
});

export default connect(
  mapStateToProps,
  { getHistory }
)(HistoryContainer);
