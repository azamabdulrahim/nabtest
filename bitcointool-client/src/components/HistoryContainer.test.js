import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";
import { HistoryContainer } from "./HistoryContainer";

describe("HistoryContainer", () => {
  // create any initial state needed
  const initialState = {};
  // here it is possible to pass in any middleware if needed into //configureStore
  const mockStore = configureStore();
  let wrapper;

  const store = mockStore(initialState);
  let mockGetHistory;

  beforeEach(() => {
    mockGetHistory = jest.fn();
  });

  it("renders correctly when refresh is false", () => {
    wrapper = shallow(
      <HistoryContainer store={store} getHistory={mockGetHistory} />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<span>No data</span>)).toBeTruthy();
    expect(mockGetHistory.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but date and currency not supplied", () => {
    const refresh = true;
    wrapper = shallow(
      <HistoryContainer
        store={store}
        refresh={refresh}
        getHistory={mockGetHistory}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<span>No data</span>)).toBeTruthy();
    expect(mockGetHistory.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but date not supplied", () => {
    const refresh = true;
    const currency = "BTC";
    wrapper = shallow(
      <HistoryContainer
        store={store}
        refresh={refresh}
        currency={currency}
        getHistory={mockGetHistory}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<span>No data</span>)).toBeTruthy();
    expect(mockGetHistory.mock.calls.length).toBe(0);
  });

  it("renders correctly when refresh is true but currency not supplied", () => {
    const refresh = true;
    const date = "20180507";
    wrapper = shallow(
      <HistoryContainer
        store={store}
        refresh={refresh}
        date={date}
        getHistory={mockGetHistory}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("div").length).toEqual(1);
    expect(wrapper.containsMatchingElement(<span>No data</span>)).toBeTruthy();
    expect(mockGetHistory.mock.calls.length).toBe(0);
  });

  it("calls getHistory function and renders child components when refresh is true", () => {
    const mockGetHistory = jest.fn();
    const refresh = true;
    const currency = "BTC";
    const date = "20180507";
    const history = {
      history: [
        {
          date: "20180705",
          currency: "BTC",
          quotes: [
            {
              time: "09:30 AM",
              price: "2.30"
            }
          ]
        },
        {
          date: "07/05/2018",
          currency: "ETC",
          quotes: [
            {
              time: "08:30 AM",
              price: "1.30"
            },
            {
              time: "12:30 PM",
              price: "1.50"
            }
          ]
        }
      ]
    };

    wrapper = shallow(
      <HistoryContainer
        store={store}
        refresh={refresh}
        date={date}
        currency={currency}
        getHistory={mockGetHistory}
        history={history}
      />
    );
    expect(wrapper).toMatchSnapshot();
    expect(mockGetHistory).toHaveBeenCalled();
    expect(wrapper.find("HistoryTable").length).toEqual(1);
  });
});
