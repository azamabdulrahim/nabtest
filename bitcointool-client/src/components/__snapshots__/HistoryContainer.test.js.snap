// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryContainer calls getHistory function and renders child components when refresh is true 1`] = `
<div>
  <p />
  <h2>
    Price History
  </h2>
  <HistoryTable
    history={
      Array [
        Object {
          "currency": "BTC",
          "date": "20180705",
          "quotes": Array [
            Object {
              "price": "2.30",
              "time": "09:30 AM",
            },
          ],
        },
        Object {
          "currency": "ETC",
          "date": "07/05/2018",
          "quotes": Array [
            Object {
              "price": "1.30",
              "time": "08:30 AM",
            },
            Object {
              "price": "1.50",
              "time": "12:30 PM",
            },
          ],
        },
      ]
    }
  />
</div>
`;

exports[`HistoryContainer renders correctly when refresh is false 1`] = `
<div>
  <p />
  <h2>
    Price History
  </h2>
  <span>
    No data
  </span>
</div>
`;

exports[`HistoryContainer renders correctly when refresh is true but currency not supplied 1`] = `
<div>
  <p />
  <h2>
    Price History
  </h2>
  <span>
    No data
  </span>
</div>
`;

exports[`HistoryContainer renders correctly when refresh is true but date and currency not supplied 1`] = `
<div>
  <p />
  <h2>
    Price History
  </h2>
  <span>
    No data
  </span>
</div>
`;

exports[`HistoryContainer renders correctly when refresh is true but date not supplied 1`] = `
<div>
  <p />
  <h2>
    Price History
  </h2>
  <span>
    No data
  </span>
</div>
`;
